tap_klaviyo_custom/streams.py

```diff
@@ -1,46 +1,214 @@
+        params.update({"api_key": self.config['api_key']})
+        return params
+
+    def prepare_request(
+        self, context: Optional[dict], next_page_token: Optional[Any]
+    ) -> requests.PreparedRequest:
+        """Prepare a request object.
+
+        If partitioning is supported, the `context` object will contain the partition
+        definitions. Pagination information can be parsed from `next_page_token` if
+        `next_page_token` is not None.
+        """
+        http_method = self.rest_method
+        url: str = self.get_url(context)
+        params: dict = self.get_url_params(context)
+        request_data = self.prepare_request_payload(context, next_page_token)
+        headers = self.http_headers
+
+        request = cast(
+            requests.PreparedRequest,
+            self.requests_session.prepare_request(
+                requests.Request(
+                    method=http_method,
+                    url=url,
+                    params=params,
+                    headers=headers,
+                    json=request_data,
+                )
+            ),
+        )
+        return request
+
+    def get_url(self, context: Optional[dict]) -> str:
+        """Return a URL, optionally targeted to a specific partition or context.
+
+        Developers override this method to perform dynamic URL generation.
+        """
+        url = self.url_base+self.path
+
+        return url
+
+    def request_records(self, context: Optional[dict]) -> Iterable[dict]:
+        """Request records from REST endpoint(s), returning response records.
+
+        If pagination is detected, pages will be recursed automatically.
+        """
+        next_page_token: Any = None
+        finished = False
+        while not finished:
+            prepared_request = self.prepare_request(
+                context, next_page_token=next_page_token
+            )
+            resp = self._request_with_backoff(prepared_request, context)
+            for row in self.parse_response(resp):
+                yield row
+            previous_token = copy.deepcopy(next_page_token)
+            next_page_token = self.get_next_page_token(
+                response=resp, previous_token=previous_token
+            )
+            if next_page_token and next_page_token == previous_token:
+                raise RuntimeError(
+                    f"Loop detected in pagination. "
+                    f"Pagination token {next_page_token} is identical to prior token."
+                )
+            # Cycle until get_next_page_token() no longer returns a value
+            finished = not next_page_token
+
+
+    def get_records(self, context: Optional[dict]) -> Iterable[Dict[str, Any]]:
+        """Return a generator of row-type dictionary objects.
+
+        Each row emitted should be a dictionary of property names to their values.
+        """
+        for row in self.request_records(context):
+            row = self.post_process(row, context)
```

Comment:
Are you overriding the post_process() function anywhere? if not, then I'd say don't need to include it. It is for you to make any cleaning (if at all) if the received record row needs it.

Reply:
I disagree. The fact that `post_process` does nothing at the moment is okay, as it's standard tap boilerplate it's already ready to go for when we need it.